Lua/LuaAdapters.cpp

```diff
@@ -287,6 +287,42 @@ namespace RTE {
 		return -1;
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void LuaAdaptersScene::CalculatePathAsync2(Scene *luaSelfObject, const luabind::object &callbackParam, const Vector &start, const Vector &end, bool movePathToGround, float digStrength, Activity::Teams team) {
+		team = std::clamp(team, Activity::Teams::NoTeam, Activity::Teams::TeamFour);
+		
+		// >:(
+		// So, we're copying into a newed ptr and passing ownership down through these lambdas instead of copying.
+		// That's because - for some god-knows-why reason - capturing the callback (by value) into the lambdas would eventually fuck up the lua state and cause random crashes elsewhere
+		// Even if we did literally nothing with it except capture it into a no-op lambda.
+		// Doing this seems to fix it. And, well, turns 3 copies into 1. So I guess that's good.
+		// Anyways, touch this at your peril. It's scary shit.
+		luabind::object* callback = new luabind::object(callbackParam);
+
+		auto callLuaCallback = [callback, movePathToGround](std::shared_ptr<volatile PathRequest> pathRequestVol) {
+			// This callback is called from the async pathing thread, so we need to further delay this logic into the main thread (via AddLuaScriptCallback)
```

Comment:
It's absolutely necessary, in case the main thread is currently running any Lua scripts (which can happen at unpredictable times due to update orders and script callbacks for Destroy(), Create() etc)